sys/riscv/riscv/machdep.c

```diff
@@ -363,6 +367,14 @@ set_mcontext(struct thread *td, mcontext_t *mcp)
 
 	tf = td->td_frame;
 
+	/*
+	 * Make sure the processor mode has not been tampered with and
+	 * interrupts have not been disabled.
+	 * Supervisor interrupts in user mode are always enabled.
+	 */
+	if ((mcp->mc_gpregs.gp_sstatus & SSTATUS_SPP) != 0)
```

Comment:
This is only checking the previous privilege, and not the previous interrupts enabled bit (SPIE). Do we also want to sanity-check SIE? (Presumably bad things happen if we take an interrupt between restoring sstatus and sret-ing..)

Reply:
Yes, that was noticed in the upstream review.  I'm going to fix that in a separate commit upstream, this one was just moving the code to the right place.